utils/python/CIME/XML/entry_id.py

```diff
@@ -151,36 +182,71 @@ def get_elements_from_child_content(self, childname, childcontent):
         nodes = self.get_nodes("entry")
         elements = []
         for node in nodes:
-            childnode = self.get_node(childname,root=node)
+            childnode = self.get_optional_node(childname,root=node)
+            expect(childnode is not None,"No childname %s for id %s"%(childname,node.get("id")))
             content = childnode.text
             if content == childcontent:
-                elements.append(node)
+                elements.append(deepcopy(node))
 
         return elements
 
-    def add_elements_by_group(self, srcobj, attlist, infile):
+    def add_elements_by_group(self, srcobj, attributes={}, infile=None):
+        """
+        Add elements from srcobj to self under the appropriate
+        group element, entries to be added must have a child element
+        <file> with value "infile"
+        """
+        if infile is None:
+            infile = os.path.basename(self.filename)
+        # First get the list of entries in srcobj with matching file children
+
         nodelist = srcobj.get_elements_from_child_content('file', infile)
-        for node in nodelist:
-            gnode = node.find(".//group")
+
+        # For matchs found: Remove {<group>, <file>, <values>}
+        # children from each entry and set the default value for the
+        # new entries in self - putting the entries as children of
+        # group elements in file $file
+        for src_node in nodelist:
+	    node  = deepcopy(src_node)
+            gnode = src_node.find(".//group")
             gname = gnode.text
+            if gname is None:
+                gname = "group_not_set"
+	    # If group with id=$gname does not exist in self.groups
+	    # then create the group node and add it to infile file
             if gname not in self.groups.keys():
                 newgroup = ET.Element("group")
                 newgroup.set("id",gname)
-                self.add_child(newgroup)
+                # initialize an empty list
                 self.groups[gname] = newgroup
+                self.add_child(newgroup)
 
-            self.set_default_value(node, attlist)
+            # Remove {<group>, <file>, <values>} from the entry element
             node = self.cleanupnode(node)
+
+	    # Add the entry element to the group
             self.groups[gname].append(node)
+
+	    # Set the default value, it may be determined by a regular
```

Comment:
We need to get the default_value from the srcnode since it may be stripped from the case node.